solarkraft/src/instrument.ts

```diff
@@ -228,24 +240,55 @@ export function isTlaName(name: string): boolean {
  *   { "a": 3, "b": 5 }  ~~>  [ a |-> 3, b |-> 5 ]
  *   { "2": 3, "4": 5 }  ~~>  SetAsFun({ <<"2", 3>>, <<"4", 5>> })
  */
-export function tlaJsonOfNative(v: any, forceVec: boolean = false): any {
+export function tlaJsonOfNative(
+    v: any,
+    forceVec: boolean = false,
+    hint?: any
+): any {
     if (typeof v === 'object') {
         if (Array.isArray(v)) {
             // a JS array
+            // we require a hint in the case of ambiguous inputs
+            const mustHaveHint =
+                !forceVec &&
+                1 <= v.length &&
+                v.length <= 2 &&
+                typeof v[0] === 'string' &&
+                (v.length === 1 || typeof v[1] === 'string')
+
+            if (mustHaveHint && hint === undefined)
+                throw new TypeError(
+                    `Ambiguous type detected for ${v}. Please \`fetch\` with --typemap provided.`
+                )
+
             if (
                 v.length == 0 ||
                 forceVec ||
-                v.every((elem) => typeof elem === typeof v[0])
+                (!mustHaveHint &&
+                    v.every((elem) => typeof elem === typeof v[0])) ||
+                (mustHaveHint && hint !== 'ENUM')
             ) {
                 // a Soroban `Vec`
                 // [ 1, 2, 3 ]  ~~>  << 1, 2, 3 >>
+                const childHints: any[] =
+                    hint === undefined
+                        ? Array(v.length).fill(undefined)
+                        : 'vec' in hint
+                          ? Array(v.length).fill(hint['vec'])
+                          : hint
```

Comment:
Should this be `hint` or `undefined`?

Reply:
I figured at that point it didn't matter, but if anything it'd probably be best to use `Array(v.length).fill(undefined)` so you don't accidentally pass unintended hints.